pkg/util/kubernetes/hpa/hpa.go

```diff
@@ -147,9 +155,6 @@ func (c *HPAWatcherClient) Start() {
 					if err = c.processHPAs(added, modified); err != nil {
 						log.Errorf("Could not update the external metrics in the store: %s", err.Error())
 					}
-					if err = c.removeEntryFromStore(deleted); err != nil {
-						log.Errorf("Could not delete the external metrics in the store: %s", err.Error())
-					}
 				}
 			// Ticker to run the refresh process for the stored external metrics
 			case <-tickerHPARefreshProcess.C:
```

Comment:
Personal note: This ticker freq has to be lower than the update from the common store.
`hpa_watcher_polling_freq` is at 10s, and `external_metrics_provider.polling_freq` is at 30s so it's good.